cmake/cc_rules.cmake

```diff
@@ -82,12 +84,30 @@ function(cc_binary name)
   set(multi DEPENDS SOURCES)
   cmake_parse_arguments(ccb "" "" "${multi}" ${ARGN})
 
-  add_executable(${name} ${ccb_SOURCES})
+  remove_objc_sources(sources ${ccb_SOURCES})
+  add_executable(${name} ${sources})
+  add_objc_flags(${name} ccb)
 
   target_include_directories(${name} PUBLIC ${FIREBASE_SOURCE_DIR})
   target_link_libraries(${name} ${ccb_DEPENDS})
 endfunction()
 
+# remove_objc_sources(output_var sources...)
+#
+# Removes Objective-C/C++ sources from the given sources if not on an Apple
```

Comment:
Nit: the function name makes it sound as if the sources were removed unconditionally. I don't have a really good suggestion, but just to brainstorm, a couple strawmen: `maybe_remove_objc_sources`, `prune_unsupported_sources`. If you don't like any of them, I'm okay if you keep as is.

Reply:
Done: `maybe_remove_objc_sources`.